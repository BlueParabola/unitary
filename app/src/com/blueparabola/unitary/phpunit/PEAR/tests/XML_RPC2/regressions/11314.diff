001- <!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
001+ Fatal error: Uncaught exception 'XML_RPC2_Exception' with message 'XML_RPC2_Backend_Php does not support any encoding other than utf-8, due to a simplexml limitation' in /media/Elements_/backup/home/clockwerx/pear-svn/packages/XML_RPC2/trunk/XML/RPC2/Backend/Php/Server.php:82
002- <html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en">
002+ Stack trace:
003-   <head>
003+ #0 /media/Elements_/backup/home/clockwerx/pear-svn/packages/XML_RPC2/trunk/XML/RPC2/Server.php(236): XML_RPC2_Backend_Php_Server->__construct(Object(XML_RPC2_Server_Callhandler_Class), Array)
004-     <meta http-equiv="Content-Type" content="text/HTML; charset=ISO-8859-1"  />
004+ #1 /media/Elements_/backup/home/clockwerx/pear-svn/packages/XML_RPC2/trunk/tests/XML_RPC2/regressions/11314.php(54): XML_RPC2_Server::create('DocumentationSe...', Array)
005-     <title>Available XMLRPC methods for this server</title>
005+ #2 {main}
006-     <style type="text/css">
006+   thrown in /media/Elements_/backup/home/clockwerx/pear-svn/packages/XML_RPC2/trunk/XML/RPC2/Backend/Php/Server.php on line 82
007-       li,p { font-size: 10pt; font-family: Arial,Helvetia,sans-serif; }
008-       a:link { background-color: white; color: blue; text-decoration: underline; font-weight: bold; }
009-       a:visited { background-color: white; color: blue; text-decoration: underline; font-weight: bold; }
010-       table { border-collapse:collapse; width: 100% }
011-       table,td { padding: 5px; border: 1px solid black; }
012-       div.bloc { border: 1px dashed gray; padding: 10px; margin-bottom: 20px; }
013-       div.description { border: 1px solid black; padding: 10px; }
014-       span.type { background-color: white; color: gray; font-weight: normal; }
015-       span.paratype { background-color: white; color: gray; font-weight: normal; }
016-       span.name { background-color: white; color: #660000; }
017-       span.paraname { background-color: white; color: #336600; }
018-       img { border: 0px; }
019-       li { font-size: 12pt; }
020-     </style>
021-   </head>
022-   <body>
023-     <h1>Available XMLRPC methods for this server</h1>
024-     <h2><a name="index">Index</a></h2>
025-     <ul>
026-       <li><a href="#2d6b7f96be69b46a6523f48b4a288864">test.getSomething()</a></li>
027-     </ul>
028-     <h2>Details</h2>
029-     <div class="bloc">
030-       <h3><a name="2d6b7f96be69b46a6523f48b4a288864"><span class="type">(string)</span> <span class="name">test.getSomething</span><span class="other">(</span><span class="paratype">(array) </span><span class="paraname">something</span>, <span class="paratype">(string) </span><span class="paraname">another_thing</span>, <span class="paratype">(boolean) </span><span class="paraname">credentials</span><span class="other">)</span></a></h3>
031-       <p><b>Description :</b></p>
032-       <div class="description">
033-         returns something
034-       </div>
035-       <p><b>Parameters : </b></p>
036-       <table>
037-         <tr><td><b>Type</b></td><td><b>Name</b></td><td><b>Documentation</b></td></tr>
038-         <tr><td>array</td><td>something</td><td>A description</td></tr>
039-         <tr><td>string</td><td>another_thing</td><td>A description of another thing</td></tr>
040-         <tr><td>boolean</td><td>credentials</td><td>Whether to return nothing - server doesn't care though</td></tr>
041-       </table>
042-       <p>(return to <a href="#index">index</a>)</p>
043-     </div>
044-   </body>
045- </html>