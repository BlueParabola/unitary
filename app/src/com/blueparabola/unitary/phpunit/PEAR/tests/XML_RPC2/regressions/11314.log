
---- EXPECTED OUTPUT
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en">
  <head>
    <meta http-equiv="Content-Type" content="text/HTML; charset=ISO-8859-1"  />
    <title>Available XMLRPC methods for this server</title>
    <style type="text/css">
      li,p { font-size: 10pt; font-family: Arial,Helvetia,sans-serif; }
      a:link { background-color: white; color: blue; text-decoration: underline; font-weight: bold; }
      a:visited { background-color: white; color: blue; text-decoration: underline; font-weight: bold; }
      table { border-collapse:collapse; width: 100% }
      table,td { padding: 5px; border: 1px solid black; }
      div.bloc { border: 1px dashed gray; padding: 10px; margin-bottom: 20px; }
      div.description { border: 1px solid black; padding: 10px; }
      span.type { background-color: white; color: gray; font-weight: normal; }
      span.paratype { background-color: white; color: gray; font-weight: normal; }
      span.name { background-color: white; color: #660000; }
      span.paraname { background-color: white; color: #336600; }
      img { border: 0px; }
      li { font-size: 12pt; }
    </style>
  </head>
  <body>
    <h1>Available XMLRPC methods for this server</h1>
    <h2><a name="index">Index</a></h2>
    <ul>
      <li><a href="#2d6b7f96be69b46a6523f48b4a288864">test.getSomething()</a></li>
    </ul>
    <h2>Details</h2>
    <div class="bloc">
      <h3><a name="2d6b7f96be69b46a6523f48b4a288864"><span class="type">(string)</span> <span class="name">test.getSomething</span><span class="other">(</span><span class="paratype">(array) </span><span class="paraname">something</span>, <span class="paratype">(string) </span><span class="paraname">another_thing</span>, <span class="paratype">(boolean) </span><span class="paraname">credentials</span><span class="other">)</span></a></h3>
      <p><b>Description :</b></p>
      <div class="description">
        returns something
      </div>
      <p><b>Parameters : </b></p>
      <table>
        <tr><td><b>Type</b></td><td><b>Name</b></td><td><b>Documentation</b></td></tr>
        <tr><td>array</td><td>something</td><td>A description</td></tr>
        <tr><td>string</td><td>another_thing</td><td>A description of another thing</td></tr>
        <tr><td>boolean</td><td>credentials</td><td>Whether to return nothing - server doesn't care though</td></tr>
      </table>
      <p>(return to <a href="#index">index</a>)</p>
    </div>
  </body>
</html>
---- ACTUAL OUTPUT
Fatal error: Uncaught exception 'XML_RPC2_Exception' with message 'XML_RPC2_Backend_Php does not support any encoding other than utf-8, due to a simplexml limitation' in /media/Elements_/backup/home/clockwerx/pear-svn/packages/XML_RPC2/trunk/XML/RPC2/Backend/Php/Server.php:82
Stack trace:
#0 /media/Elements_/backup/home/clockwerx/pear-svn/packages/XML_RPC2/trunk/XML/RPC2/Server.php(236): XML_RPC2_Backend_Php_Server->__construct(Object(XML_RPC2_Server_Callhandler_Class), Array)
#1 /media/Elements_/backup/home/clockwerx/pear-svn/packages/XML_RPC2/trunk/tests/XML_RPC2/regressions/11314.php(54): XML_RPC2_Server::create('DocumentationSe...', Array)
#2 {main}
  thrown in /media/Elements_/backup/home/clockwerx/pear-svn/packages/XML_RPC2/trunk/XML/RPC2/Backend/Php/Server.php on line 82
---- FAILED
