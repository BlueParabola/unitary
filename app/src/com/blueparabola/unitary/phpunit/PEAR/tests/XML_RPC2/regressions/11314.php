<?php
set_include_path(realpath(dirname(__FILE__) . '/../../../') . PATH_SEPARATOR . get_include_path());
/**
 * Point to a problem with the autodocumentation of servers which follows the specifications
 * in PHPCodeSniffer.
 *
 * PHP version 5
 *
 * @category  XML
 * @package   XML_RPC2
 * @author    Lars Olesen <lars@legestue.net>

 * @copyright 2007 Lars Olesen
 * @license   GPL http://www.opensource.org/licenses/gpl-license.php
 * @version   @package-version@
 * @link      http://pear.php.net/package/XML_RPC2
 */
require_once 'XML/RPC2/Server.php';

/**
 * The implementation
 *
 * @category  XML
 * @package   XML_RPC2
 * @author    Lars Olesen <lars@legestue.net>
 * @copyright 2007 Lars Olesen
 * @license   GPL http://www.opensource.org/licenses/gpl-license.php
 * @version   @package-version@
 * @link      http://pear.php.net/package/XML_RPC2
 */
class DocumentationServer {

    /**
     * returns something
     *
     * @param array   $something     A description
     * @param string  $another_thing A description of another thing
     * @param boolean $return        Whether to return nothing - server doesn't care though
     *
     * @return string An international string
     */
    public static function getSomething($something, $another_thing, $credentials) {
        return 'nothing interesting';
    }

}

$options = array(
    'prefix' => 'test.',
  //  'encoding' => 'ISO-8859-1'
  'encoding' => 'UTF-8'
);

$server = XML_RPC2_Server::create('DocumentationServer', $options);
$GLOBALS['HTTP_RAW_POST_DATA'] = '';
$server->handleCall();
?>
