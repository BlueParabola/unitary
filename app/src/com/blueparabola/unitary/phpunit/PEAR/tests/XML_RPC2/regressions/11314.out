Fatal error: Uncaught exception 'XML_RPC2_Exception' with message 'XML_RPC2_Backend_Php does not support any encoding other than utf-8, due to a simplexml limitation' in /media/Elements_/backup/home/clockwerx/pear-svn/packages/XML_RPC2/trunk/XML/RPC2/Backend/Php/Server.php:82
Stack trace:
#0 /media/Elements_/backup/home/clockwerx/pear-svn/packages/XML_RPC2/trunk/XML/RPC2/Server.php(236): XML_RPC2_Backend_Php_Server->__construct(Object(XML_RPC2_Server_Callhandler_Class), Array)
#1 /media/Elements_/backup/home/clockwerx/pear-svn/packages/XML_RPC2/trunk/tests/XML_RPC2/regressions/11314.php(54): XML_RPC2_Server::create('DocumentationSe...', Array)
#2 {main}
  thrown in /media/Elements_/backup/home/clockwerx/pear-svn/packages/XML_RPC2/trunk/XML/RPC2/Backend/Php/Server.php on line 82